controllers/consumer_controller.go

```diff
@@ -215,48 +132,41 @@ func (r *ConsumerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 		log.V(1).Info("created new deployment", "deployment", newD, "partition", partition)
 		r.Recorder.Eventf(
-			&consumer,
+			co.consumer,
 			corev1.EventTypeNormal,
 			"DeployCreate",
 			"deployment for partition was created %d", partition,
 		)
 	}
 
-	for _, deploy := range toRemoveInstances {
+	for _, deploy := range co.toRemoveInstances {
 		if err := r.Delete(ctx, deploy); errors.IgnoreNotFound(err) != nil {
 			log.Error(err, "unable to delete deployment", "deployment", deploy)
 		}
 		r.Recorder.Eventf(
-			&consumer,
+			co.consumer,
 			corev1.EventTypeNormal,
 			"DeployDelete",
 			"deployment %s was deleted", deploy.Name,
 		)
 	}
 
-	for _, origDeploy := range toUpdateInstances {
+	updateDeploy := func(origDeploy *appsv1.Deployment) {
```

Comment:
should it be a method of `co`?

Reply:
also, down the road, behavior of `toUpdateInstances` and `toEstimateInstances` will be a bit different

Reply:
> should it be a method of co?

it will require `ConsumerReconciler` as dependency then

> also, down the road, behavior of toUpdateInstances and toEstimateInstances will be a bit different

no problems, not a fan of that wrapper) should I revert it to two sep actions?

Reply:
it feels like it makes more sense to have it separate